server/src/test/java/io/crate/lucene/BooleanEqQueryTest.java

```diff
@@ -107,4 +107,15 @@ public void test_BooleanEqQuery_TermsQuery() {
         assertThat(query.getClass().getName()).endsWith("SortedNumericDocValuesSetQuery");
         assertThat(query).hasToString("arr2: [0, 1]");
     }
+
+    @Test
+    public void test_not_on_boolean_types() {
+        Query query = convert("not(a1)");
+        assertThat(query).isExactlyInstanceOf(BooleanQuery.class);
+        assertThat(query).hasToString("+(+*:* -a1:T) #(NOT a1)");
```

Comment:
The Lucene query can be reduced to `a1:F` will create a separate issue for this.

Reply:
I can expand on this PR (as a follow up) by adding more tests like `neq`, `not range query` and also to different types.

Reply:
Also look for more edge scenarios